core/views.py

```diff
@@ -316,10 +317,17 @@ def form_valid(self, form):
         jobs = form.cleaned_data['jobs'].split(',')
 
         highlights = form.cleaned_data['highlights']
+        highlights_list = json.loads(highlights)
 
-        if highlights != '':
-            program.highlights = highlights
+        # Filter out entries with empty 'icon_class' or 'description'
+        filtered_highlights = [entry for entry in highlights_list if entry['icon_class'] or entry['description']]
+        if filtered_highlights:
+            program.highlights = json.dumps(filtered_highlights)
             program.save()
+        if not filtered_highlights:
+            program.highlights = []
+            program.save()
```

Comment:
Have we verified this is working? On line 325 above, we're calling `json.dumps()`, which takes the object and converts to a JSON formatted string. In this case, we're passing in a python list. There may be some magic under the hood that will automatically pass the object through `json.dumps()` on the JSON field, but I don't know that for sure. We should at least check to make sure this works as expected.

Reply:
Or alternatively, we can just update this to:

```suggestion
            program.highlights = json.dumps([])
            program.save()
```